flashlight: Add support for simpler sysfs toggles

JamiKettunen opened the pull request:
This is used by the Volla Phone (Helio P23 / MT6763) for example.

**HELP WANTED:** What `sysfs` paths do legacy / other existing ported MTK devices use for flashlight control? We could add support for them in this PR as well :)

NotKit commented on src/flashlight.c:
```diff
@@ -26,25 +26,37 @@
 
 char* flash_sysfs_path = NULL;
 gboolean activated = 0;
+gboolean is_qcom_torch = 0;
```
I think enum type variable might be a better idea for the case we ever need to extend it.

NotKit commented on src/flashlight.c:
```diff
@@ -82,9 +94,9 @@ toggle_flashlight_action(GAction *action,
         fprintf(fd2, "0");
       else
-        fprintf(fd1, QCOM_DISABLE);
+        fprintf(fd1, (is_qcom_torch ? QCOM_DISABLE : MTK_DISABLE));
```
Then based on vendor variable execution path can be split into toggle_flashlight_action_qcom/toggle_flashlight_action_mtk. This may lead to a bit of code duplication, but should be much easier to read/understand what is going on.

NotKit commented:
Just tested, on PinePhone flashlight can be controlled by writing 0/1 to /sys/class/leds/white:flash/brightness, can be added to handling as well.

JamiKettunen committed:
flashlight: Add support for PinePhone's sysfs

This sysfs path was tested as working by NotKit.

JamiKettunen committed:
flashlight: Add support for simpler sysfs toggles

Unlike the overcomplicated Qualcomm mess, there are also simpler sysfs
toggles which you can just write 1/0 to when enabling/disabling the
flash.

This is used by the MediaTek flashlight_core driver on the Volla Phone
(Helio P23 / MT6763) for example, which this also adds support for.

dobey reviewed (approved):
Looks OK. Thanks.

ghost commented:
I tested this on the PinePhone, but it doesn't seem to do anything. Manually writing to `/sys/class/leds/white:flash/brightness` does work though.

JamiKettunen commented:
> I tested this on the PinePhone, but it doesn't seem to do anything. Manually writing to sudo `/sys/class/leds/white:flash/brightness` does work though.

I don't personally own a PinePhone, but this could be a permission issue as I believe the power indicator doesn't run as `root`; on Volla Phone (`flashlight_core` sysfs) the control file seems to have permissions of `a+rw` (definitely not ideal, but hey that's been the default in my case)

ghost commented:
`chmod 666` indeed fixed it. Let's see if I can make a udev rule for that..

ghost commented:
https://gitlab.com/ubports/community-ports/pinephone/-/merge_requests/9
